pkg/controller/core.oam.dev/v1alpha2/application/generator.go

```diff
@@ -181,6 +187,9 @@ func (h *AppHandler) applyComponentFunc(appParser *appfile.Parser, appRev *v1bet
 		if !isHealth {
 			return nil, nil, false, nil
 		}
+		if optimize.WorkflowOptimizer.DisableResourceApplyDoubleCheck {
+			return readyWorkload, readyTraits, true, nil
+		}
```

Comment:
as we disable the double check, can we still ensure the resources is created in health check?

Reply:
No, if health policy is configured, we can. Otherwise not.